src/common/watchdog/systemd/watchdog.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/watchdog.hpp>
+
+#include <systemd/sd-daemon.h>
+
+#include <common/log.hpp>
+
+namespace mender {
+namespace common {
+namespace watchdog {
+
+using namespace std;
+
+void Kick() {
+  log::Trace("Kicking the application watchdog");
+	uint64_t usec;
+	if (sd_watchdog_enabled(0, &usec) > 1) {
+		int ret {sd_notify(0 /* Do not unset NOTIFY_SOCKET env var */, "WATCHDOG=1")};
+		if (ret < 0) {
+			log::Error(
+				"Failed to kick the systemd service watchdog, received error code: "
+				+ to_string(ret));
+		}
+		if (ret == 0) {
+			log::Error(
+				"The NOTIFY_SOCKET environment variable is not set, so it is not possible to kick the watchdog");
```

Comment:
Can this even happen when `sd_watchdog_enabled()` returns a positive value?

Reply:
Probably not, unless something weird happens in the meantime.

Defense, or clean code? :thinking:

Reply:
haha, alright. Sold!